From 299b1ff20a7d224bc818f1f4857d67557389fbcc Mon Sep 17 00:00:00 2001
From: He Chuyue <hechuyue@wxiat.com>
Date: Tue, 7 Jun 2022 17:42:35 +0800
Subject: [PATCH] sw64: remove unnecessary parameter in REG_OFFSET_NAME
Patch-mainline: NO, OTHERS
References: openEuler-22.03-LTS
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

The `reg` and `r` are always the same, so save one of them and
remove the other.

Signed-off-by: He Chuyue <hechuyue@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/ptrace.c | 63 +++++++++++++++++++-------------------
 1 file changed, 32 insertions(+), 31 deletions(-)

diff --git a/arch/sw_64/kernel/ptrace.c b/arch/sw_64/kernel/ptrace.c
index 649572bbe613..b06c98e9944b 100644
--- a/arch/sw_64/kernel/ptrace.c
+++ b/arch/sw_64/kernel/ptrace.c
@@ -608,8 +608,8 @@ struct pt_regs_offset {
 	int offset;
 };
 
-#define REG_OFFSET_NAME(reg, r) {				\
-	.name = #reg,						\
+#define REG_OFFSET_NAME(r) {					\
+	.name = #r,						\
 	.offset = offsetof(struct pt_regs, r)			\
 }
 
@@ -619,37 +619,38 @@ struct pt_regs_offset {
 }
 
 static const struct pt_regs_offset regoffset_table[] = {
-	REG_OFFSET_NAME(r0, r0),
-	REG_OFFSET_NAME(r1, r1),
-	REG_OFFSET_NAME(r2, r2),
-	REG_OFFSET_NAME(r3, r3),
-	REG_OFFSET_NAME(r4, r4),
-	REG_OFFSET_NAME(r5, r5),
-	REG_OFFSET_NAME(r6, r6),
-	REG_OFFSET_NAME(r7, r7),
-	REG_OFFSET_NAME(r8, r8),
-	REG_OFFSET_NAME(r19, r19),
-	REG_OFFSET_NAME(r20, r20),
-	REG_OFFSET_NAME(r21, r21),
-	REG_OFFSET_NAME(r22, r22),
-	REG_OFFSET_NAME(r23, r23),
-	REG_OFFSET_NAME(r24, r24),
-	REG_OFFSET_NAME(r25, r25),
-	REG_OFFSET_NAME(r26, r26),
-	REG_OFFSET_NAME(r27, r27),
-	REG_OFFSET_NAME(r28, r28),
-	REG_OFFSET_NAME(hae, hae),
-	REG_OFFSET_NAME(trap_a0, trap_a0),
-	REG_OFFSET_NAME(trap_a1, trap_a1),
-	REG_OFFSET_NAME(trap_a2, trap_a2),
-	REG_OFFSET_NAME(ps, ps),
-	REG_OFFSET_NAME(pc, pc),
-	REG_OFFSET_NAME(gp, gp),
-	REG_OFFSET_NAME(r16, r16),
-	REG_OFFSET_NAME(r17, r17),
-	REG_OFFSET_NAME(r18, r18),
+	REG_OFFSET_NAME(r0),
+	REG_OFFSET_NAME(r1),
+	REG_OFFSET_NAME(r2),
+	REG_OFFSET_NAME(r3),
+	REG_OFFSET_NAME(r4),
+	REG_OFFSET_NAME(r5),
+	REG_OFFSET_NAME(r6),
+	REG_OFFSET_NAME(r7),
+	REG_OFFSET_NAME(r8),
+	REG_OFFSET_NAME(r19),
+	REG_OFFSET_NAME(r20),
+	REG_OFFSET_NAME(r21),
+	REG_OFFSET_NAME(r22),
+	REG_OFFSET_NAME(r23),
+	REG_OFFSET_NAME(r24),
+	REG_OFFSET_NAME(r25),
+	REG_OFFSET_NAME(r26),
+	REG_OFFSET_NAME(r27),
+	REG_OFFSET_NAME(r28),
+	REG_OFFSET_NAME(hae),
+	REG_OFFSET_NAME(trap_a0),
+	REG_OFFSET_NAME(trap_a1),
+	REG_OFFSET_NAME(trap_a2),
+	REG_OFFSET_NAME(ps),
+	REG_OFFSET_NAME(pc),
+	REG_OFFSET_NAME(gp),
+	REG_OFFSET_NAME(r16),
+	REG_OFFSET_NAME(r17),
+	REG_OFFSET_NAME(r18),
 	REG_OFFSET_END,
 };
+
 /**
  * regs_query_register_offset() - query register offset from its name
  * @name:       the name of a register
-- 
2.34.1

